### PR TITLE
Shift module

### PR DESCRIPTION
**Pull request by sebastianbagya**
add id into ShiftVO 
test ShiftRepository

### CODE REVIEW

File: src/main/java/com/blibli/future/controller/ShiftController.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+public class ShiftController {
+
+    private final String BASE_PATH = "/shift";
+    private final String PATH_WITH_ID = BASE_PATH + "/{id}";
```
Comment (riencus) [inline src/main/java/com/blibli/future/controller/ShiftController.java]: After doing some "comparative study", I think it should be public so the test can reuse the path
`private final String ...` -> `public final String ...`

File: src/main/java/com/blibli/future/model/Shift.java [hunk 2]
```diff
@@ -2,22 +2,34 @@
...
 import javax.persistence.Entity;
 import javax.persistence.Id;
-import java.sql.Timestamp;
+import java.time.DayOfWeek;
+import java.time.LocalTime;
 
 @Entity
 public class Shift {
 	@Id
 	private String id;
```
Comment (riencus) [inline src/main/java/com/blibli/future/model/Shift.java:11]: If we are using an Id, isn't it better to make it updated automatically? like `EmployeeAbsencePermit`?
```
@GeneratedValue(generator = "uuid")
@GenericGenerator(name = "uuid", strategy = "uuid2")
```

File: src/test/java/com/blibli/future/controller/ShiftControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+public class ShiftControllerTest {
+    @InjectMocks
+    private ShiftController shiftController;
+    @Mock
+    private ShiftService shiftService;
+
+    private MockMvc mockMvc;
+
+    private final String BASE_PATH = "/shift";
+    private final String PATH_WITH_ID = BASE_PATH + "/test";
```
Comment (riencus) [inline src/test/java/com/blibli/future/controller/ShiftControllerTest.java:36]: My "comparative study" said that we don't need to write this. Just use the one declared as public in the Controller

File: src/test/java/com/blibli/future/controller/ShiftControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    private final String TEST = "test";
+
+    private Shift shift1;
+    private Shift shift2;
+    private List collectionMock;
+    private ShiftVO shiftVOMock;
+
+    ObjectWriter objectWriter = new ObjectMapper().writer();
```
Comment (riencus) [inline src/test/java/com/blibli/future/controller/ShiftControllerTest.java]: Isn't this should be declared as private?

**Review by riencus — CHANGES_REQUESTED** [review]
A little change that will make our code better (according to my "comparative study")

File: src/main/java/com/blibli/future/model/Shift.java (hunk 2 above)
Comment (sebastianbagya) [reply src/main/java/com/blibli/future/model/Shift.java:11]: are u sure? uuid will generate random text. I think its not good to be applied for non-transaction case.

File: src/test/java/com/blibli/future/controller/ShiftControllerTest.java (hunk 3 above)
Comment (sebastianbagya) [reply src/test/java/com/blibli/future/controller/ShiftControllerTest.java:36]: I only make repository test. It came from previous commit.

File: src/main/java/com/blibli/future/controller/ShiftController.java (hunk 1 above)
Comment (sebastianbagya) [reply src/main/java/com/blibli/future/controller/ShiftController.java]: I only make repository test. It came from previous commit. But thanks for review.

File: src/test/java/com/blibli/future/controller/ShiftControllerTest.java (hunk 4 above)
Comment (sebastianbagya) [reply src/test/java/com/blibli/future/controller/ShiftControllerTest.java]: I only make repository test. It came from previous commit. But thanks for review.

**Review by sebastianbagya — COMMENTED** [review]
already done

**Review by sebastianbagya — COMMENTED** [review]
please review once more

**Pull request merged**